### PR TITLE
DOC: stats: Explain meaning of alternatives for fisher_exact.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Partly motivated by work in https://github.com/scipy/scipy/pull/13340.  It will be useful to have a more thorough explanation of alternatives provided by `fisher_exact`, so it can be referred to in the `odds_ratio` docstring.

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -4095,20 +4097,105 @@ def fisher_exact(table, alternative='two-sided'):
...
+        >>> M = table.sum()
+        >>> n = table[0].sum()
+        >>> N = table[:, 0].sum()
+        >>> from scipy.stats import hypergeom
+        >>> start, end = hypergeom.support(M, n, N)
+        >>> hypergeom.pmf(np.arange(start, end+1), M, n, N)
+        array([0.01631702, 0.16317016, 0.40792541, 0.32634033, 0.08158508,
+               0.004662  ])
+
+    The two-sided p-value is the probabilty that, under the null hypothesis,
```
Comment (tylerjereddy) [inline scipy/stats/stats.py:4151]: `probabilty` -> `probability` (I was just skimming through as non-expert)

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -4095,20 +4097,105 @@ def fisher_exact(table, alternative='two-sided'):
...
 
     Notes
     -----
+    *Null hypothesis and p-values*
+
+    The null hypothesis is that the input table is from the hypergeometric
+    distribution with parameters (as used in `scipy.stats.hypergeom`)
+    ``M = a + b + c + d``, ``n = a + b`` and ``N = a + c``, where the
+    input table is ``[[a, b], [c, d]]``.  This distribution has support
+    ``max(0, N + n - M) <= x <= min(N, n)``, or, in terms of the values
```
Comment (tirthasheshpatel) [inline scipy/stats/stats.py:4113]: How about rendering this as latex blocks?

```suggestion
    :math:`\max(0, N + n - M) \le x \le \min(N, n)`, or, in terms of the values
```

File: scipy/stats/stats.py [hunk 3]
```diff
@@ -4095,20 +4097,105 @@ def fisher_exact(table, alternative='two-sided'):
...
+    in the input table, ``min(0, a - d) <= x <= a + min(b, c)``.  ``x``
+    can be interpreted as the upper-left element of a 2x2 table, so the
```
Comment (tirthasheshpatel) [inline scipy/stats/stats.py:4115]: Here too:

```suggestion
    in the input table, :math:`\min(0, a - d) \le x \le a + \min(b, c)`.
    ``x`` can be interpreted as the upper-left element of a 2x2 table, so the
```

File: scipy/stats/stats.py [hunk 4]
```diff
@@ -4095,20 +4097,105 @@ def fisher_exact(table, alternative='two-sided'):
...
+        [  x           n - x     ]
+        [N - x    M - (n + N) + x]
+
+    For example, if::
+
+        table = [6  2]
+                [1  4]
+
+    then the support is ``2 <= x <= 7``, and the tables in the distribution
+    are::
```
Comment (tirthasheshpatel) [inline scipy/stats/stats.py:4127]: Here too:

```suggestion
    then the support is :math:`2 \le x \le 7`, and the tables in the
    distribution are::
```

File: scipy/stats/stats.py (hunk 2 above)
Comment (rgommers) [reply scipy/stats/stats.py:4113]: I'd prefer these as they are to be honest. It looks okay as code in html, and the plain text version is way more readable than in LaTeX format.

**Review by rgommers — APPROVED** [review]
LGTM, thanks @WarrenWeckesser.

Pushed a tiny fix for the typo and formatting, merging now.

File: scipy/stats/stats.py (hunk 2 above)
Comment (tirthasheshpatel) [reply scipy/stats/stats.py:4117]: > It looks okay as code in html, and the plain text version is way more readable than in LaTeX format.

Oh, OK. I thought using LaTeX can help with how they look in HTML, but didn't think of readability in REPL. If that's something we should look out for, I agree this makes more sense as it is. Thanks!

Comment (rgommers) [reply scipy/stats/stats.py:4117]: Yeah, often the code/pseudo-code is awkward, and then it makes total sense to use LaTeX in the Notes section. But if it looks good as code, then I'd say prefer that.

Comment (rgommers) [reply scipy/stats/stats.py:4117]: Also in the editor - my brain needs more time to parse `\le` than `<` :)

**Pull request merged**